website/project/model.py

```diff
@@ -1128,6 +1146,36 @@ def can_comment(self, auth):
             )
         return self.is_contributor(auth.user)
 
+    def set_node_license(self, data, auth, save=True):
```

Comment:
I used `set_title` as a reference and was surprised to not see auth checks there, but given the way this method is called (now at least), auth is getting checked beforehand.

Not saying this is the right way-- maybe we should add permissions checks to `set_title` as well?